src/components/bottom_bar_statistic/aggregate_statistics_store.ts

```diff
@@ -0,0 +1,140 @@
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values) => countAny([[values]]),
+  },
+  {
+    name: _t("Count Numbers"),
+    types: [CellValueType.number, CellValueType.text, CellValueType.boolean, CellValueType.error],
+    compute: (values, locale) => countNumbers([[values]], locale),
+  },
+];
+
+export class AggregateStatisticsStore extends SpreadsheetStore {
+  statisticFnResults: StatisticFnResults = this._computeStatisticFnResults();
+  private isDirty = false;
+
+  constructor(get: Get) {
+    super(get);
+    this.model.selection.observe(this, {
+      handleEvent: this.handleEvent.bind(this),
+    });
+  }
+
+  handle(cmd: Command) {
+    if (
+      invalidateEvaluationCommands.has(cmd.type) ||
+      (cmd.type === "UPDATE_CELL" && "content" in cmd)
+    ) {
+      this.isDirty = true;
+    }
+    switch (cmd.type) {
+      case "HIDE_COLUMNS_ROWS":
+      case "UNHIDE_COLUMNS_ROWS":
+      case "GROUP_HEADERS":
+      case "UNGROUP_HEADERS":
+      case "ACTIVATE_SHEET":
+      case "ACTIVATE_NEXT_SHEET":
+      case "ACTIVATE_PREVIOUS_SHEET":
+      case "EVALUATE_CELLS":
+      case "UNDO":
+      case "REDO":
+        this.isDirty = true;
+    }
+  }
+
+  finalize() {
+    if (this.isDirty) {
+      this.isDirty = false;
+      this.statisticFnResults = this._computeStatisticFnResults();
+    }
+  }
+
+  handleEvent() {
+    if (this.getters.isGridSelectionActive()) {
+      this.statisticFnResults = this._computeStatisticFnResults();
+    }
+  }
+
+  private _computeStatisticFnResults(): StatisticFnResults {
+    const getters = this.getters;
+    const sheetId = getters.getActiveSheetId();
+    const cells = new Set<EvaluatedCell>();
+
+    const zones = getters.getSelectedZones();
+    for (const zone of zones) {
+      for (let col = zone.left; col <= zone.right; col++) {
+        for (let row = zone.top; row <= zone.bottom; row++) {
+          if (getters.isRowHidden(sheetId, row) || getters.isColHidden(sheetId, col)) {
+            continue; // Skip hidden cells
+          }
+
+          const evaluatedCell = getters.getEvaluatedCell({ sheetId, col, row });
+          if (evaluatedCell.type !== CellValueType.empty) {
+            cells.add(evaluatedCell);
+          }
+        }
+      }
+    }
+    const locale = getters.getLocale();
+    let statisticFnResults: StatisticFnResults = {};
+    for (let fn of selectionStatisticFunctions) {
+      // We don't want to display statistical information when there is no interest:
+      // We set the statistical result to undefined if the data handled by the selection
+      // does not match the data handled by the function.
+      // Ex: if there are only texts in the selection, we prefer that the SUM result
+      // be displayed as undefined rather than 0.
+      let fnResult: number | undefined = undefined;
+      const evaluatedCells = [...cells].filter((c) => fn.types.includes(c.type));
```

Comment:
the [...cells] could be done once outside the loop, `filter` returns a shallow copy so it can be used multiple times on the same structure